models/observations.js

```diff
@@ -65,5 +65,11 @@ module.exports = {
       type : Boolean,
       index: true
     }
-  }
+  },
+  compoundIndex: [
+    {
+        "name" :{ createdBy: 1, solutionId: 1 },
+        "indexType" : { unique: true, partialFilterExpression: { solutionId: { $exists: true }}}
```

Comment:
@priyanka-TL there will never be a case where solutionId is not there right?

Reply:
no